pkg/apis/build/v1alpha1/parameter.go

```diff
@@ -4,9 +4,9 @@
 
 package v1alpha1
 
-// Parameter defines the data structure that would be used for
-// expressing arbitrary key/value pairs for the execution of a build
-type Parameter struct {
+// Param is a key/value that populates a strategy parameter
+// used in the execution of the strategy steps
+type Param struct {
```

Comment:
How do you feel about `ParamValue`, a field named `paramValues:`, etc.? It's a bit confusing to have both `Parameter` and `Param` types (Tekton has this issue as well)

Reply:
not my favorite field name tbh, but I see the benefit. The concern I have now, is that this naming conventions were already accepted via the [related EP](https://github.com/shipwright-io/build/blob/main/docs/proposals/parameterize-strategies.md) before.

My options on this:

- Block this till consensus from more folks, as the change `params`->`paramValues` is high.
- Merge this, and then look for consensus. If we need to change this, then we PR the change. 
 
What do you think?

Reply:
@imjasonh I did the changes as requested. FYI I sync on this offline with @SaschaSchwarze0 and @HeavyWombat , we all agree its a good alternative. Please re-review this PR again.